### PR TITLE
KubernetesV1: Sanitize kubectl version output to prevent ##vso[ comma…

### PR DESCRIPTION
**Pull request by manolerazvan**
### **Context**
Security hardening for the KubernetesV1 pipeline task. The `displayKubectlVersion()` function logs output from `kubectl version` without sanitizing for Azure Pipelines logging-command patterns, which could allow specially crafted server responses to be interpreted as agent commands.

---

### **Task Name**
KubernetesV1

---

### **Description**
- Extracted `sanitizeForLoggingCommand()` into `src/sanitize.ts` — strips untrusted strings before they are written to stdout.
- Applied sanitization to both `clientVersion.gitVersion` and `serverVersion.gitVersion` in `displayKubectlVersion()`.
- Added 8 unit tests covering normal pass-through, injection patterns, case sensitivity, newline splitting, null/empty handling, and multiple injections.
- Bumped task version 1.275.0 → 1.276.0.

---

### **Risk Assessment** (Low / Medium / High)
**Low** — The change only affects display/logging output of version strings. No logic flow, cluster connection, or kubectl command execution is modified. The sanitization is additive (replace dangerous patterns with safe equivalents).

---

### **Change Behind Feature Flag** (Yes / No)
No — This is a security fix that must apply universally. A feature flag would leave users vulnerable. The attack surface is limited to display output only, so risk of regression is minimal.

---

### **Tech Design / Approach**
- The sanitizer replaces `##vso[` with `__vso[` and `##[` with `__[` (safe visual equivalents that the agent will not parse as commands).
- Newlines are collapsed to spaces to prevent line-splitting attacks.
- Function extracted to separate module (`src/sanitize.ts`) to enable clean unit testing without triggering heavy side effects from the main `kubernetes.ts` module.

---

### **Documentation Changes Required** (Yes/No)
No — No user-facing behavior change; version strings display identically for legitimate clusters.

---

### **Unit Tests Added or Updated** (Yes / No)
Yes — 8 new unit tests in `Tests/L0.ts` covering all sanitization scenarios. All passing.

---

### **Additional Testing Performed**
- Verified existing L0 test suite still compiles and the sanitization tests pass independently.
- Confirmed the sanitizer handles edge cases: null, undefined, empty string, multiple injections in one string.
- Manual verification that legitimate version strings (e.g., `v1.30.0`, `v1.12.7+hotfix`) pass through unchanged.

---

### **Logging Added/Updated** (Yes/No)
No — Existing logging is preserved; only the logged values are now sanitized.

---

### **Telemetry Added/Updated** (Yes/No)
No — The existing `##vso[telemetry.publish]` call in the task is unaffected (it uses hardcoded safe values, not user input).

---

### **Rollback Scenario and Process** (Yes/No)
Yes — Revert to task version 1.275.0. The change is purely additive to logging output; rolling back restores original behavior with no side effects.

---

### **Dependency Impact Assessed and Regression Tested** (Yes/No)
Yes — No new dependencies introduced. The sanitize module has zero imports. Existing tests compile and the task JSON schema is unchanged except for the version bump.

---

### **Checklist**
- [x] Related issue linked (if applicable)
- [x] Task version was bumped — see [versioning guide](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md)
- [x] Verified the task behaves as expected

### CODE REVIEW

**Review by rm-dmiri — APPROVED** [review]

**Comment by wawanawna** [comment]
LGTM, ready to merge

**Review by wawanawna — APPROVED** [review]

File: Tasks/KubernetesV1/src/sanitize.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ */
+export function sanitizeForLoggingCommand(value: string): string {
+    if (!value) {
+        return value;
+    }
+    return value
+        .replace(/##vso\[/gi, '__vso[')
+        .replace(/##\[/g, '__[')
+        .replace(/[\r\n]+/g, ' ');
+}
```
[the remark this answers is not in the document]
Comment (manolerazvan) [reply Tasks/KubernetesV1/src/sanitize.ts]: resolved

**Review by nemanjarogic — APPROVED** [review]
